### PR TITLE
fix(#32): now public access modifier is added on ts transformation level

### PR DESCRIPTION
**Pull request by VintorezVS**

### CODE REVIEW

File: tests/classes/transformClassDeclaration.test.ts [hunk 1]
```diff
@@ -80,3 +83,67 @@ class FooBarError extends Error {
     "
   `);
 });
+
+test('adds access modifier for class members', async () => {
+  await expect(ts2hx`
+  class Foo {
```
Comment (alexeychikk) [inline tests/classes/transformClassDeclaration.test.ts:89]: Please get rid of additional tabulation, check other tests. Make sure enclosing "`" is also at the start of the line.

File: src/transformers/Transpiler.ts [hunk 2]
```diff
@@ -202,7 +203,7 @@ export interface VisitNodeContext {
 
 export type TransformerFn = (
   this: Transpiler,
-  node: ts.Node,
+  node: ts.Node & { parent: ts.Node['parent'] | undefined },
```
Comment (alexeychikk) [inline src/transformers/Transpiler.ts:206]: I don't think this is necessary. Compiles even without this change.
For example, `ts.SourceFile` doesn't have parent and still ts devs didn't make it optional.

File: src/transformers/ts/objects.ts [hunk 3]
```diff
@@ -8,7 +8,7 @@ export const transformMethodOnObject: TransformerFn = function (
 ) {
   // { methodOnObject() {} }
   if (!ts.isMethodDeclaration(node)) return;
-  if (!ts.isObjectLiteralExpression(node.parent)) return;
+  if (!ts.isObjectLiteralExpression(this.utils.getParentNode(node))) return;
```
Comment (alexeychikk) [inline src/transformers/ts/objects.ts:11]: I'm curious why have you replaced `node.parent` only here and not everywhere else?

File: src/transformers/utils/common.ts [hunk 4]
```diff
@@ -194,3 +194,7 @@ export function ensureNodeIsBlock(
 
   throw new Error('Unable to transform node to Block');
 }
+
+export function getParentNode(this: Transpiler, node: ts.Node): ts.Node {
+  return node.parent ?? ts.getOriginalNode(node).parent;
```
Comment (alexeychikk) [inline src/transformers/utils/common.ts:199]: I think I understand why this is needed but I don't like this solution.
I guess I know where we got things wrong: we are using just 1 transformer here:
https://github.com/alexeychikk/ts2hx/blob/965f5f8fab46799b33654eda598e1941c3502399/src/transformers/Transpiler.ts#L61
I think what we should've done is to map our transformers to a set of `ts.TransformerFactory`.
Maybe this way ts would do parents linking for us.

**Review by alexeychikk — CHANGES_REQUESTED** [review]

File: tests/classes/transformClassDeclaration.test.ts [hunk 5]
```diff
@@ -80,3 +83,67 @@ class FooBarError extends Error {
     "
   `);
 });
+
+test('adds access modifier for class members', async () => {
```
Comment (alexeychikk) [inline tests/classes/transformClassDeclaration.test.ts:87]: 1 test file is intended for 1 particular transformer. Please extract this test into a separate file

**Review by alexeychikk — CHANGES_REQUESTED** [review]

File: src/transformers/Transpiler.ts (hunk 2 above)
Comment (VintorezVS) [reply src/transformers/Transpiler.ts:206]: ups forgot to remove this

File: src/transformers/ts/objects.ts (hunk 3 above)
Comment (VintorezVS) [reply src/transformers/ts/objects.ts:11]: It is needed only for ts transformation layer.

**Review by alexeychikk — APPROVED** [review]

**Pull request merged**